javascript-typing-tutor/index.html

```diff
@@ -0,0 +1,117 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <title>JavaScript Typing Tutor</title>
+    <link rel="stylesheet" href="styles.css" />
+  </head>
+  <body>
+    <div class="word">
+      <span class="underscore">p</span>
+      <span>l</span>
+      <span>e</span>
+      <span>a</span>
+      <span>s</span>
+      <span>e</span>
+      <span>&nbsp;</span>
+      <span>t</span>
+      <span>y</span>
+      <span>p</span>
+      <span>e</span>
+      <span>&nbsp;</span>
+      <span>t</span>
+      <span>h</span>
+      <span>i</span>
+      <span>s</span>
+      <span>&nbsp;</span>
+      <span>s</span>
+      <span>e</span>
+      <span>n</span>
+      <span>t</span>
+      <span>e</span>
+      <span>n</span>
+      <span>c</span>
+      <span>e</span>
+    </div>
+    <script>
+      const spanElements = document.querySelectorAll('span');
+      const bodySelector = document.querySelector('body');
+      let currentPosition = 0;
+      let attempts = 0;
+      let rightKeys = 0;
+      let x = true;
+
+      bodySelector.addEventListener('keydown', function (event) {
+        if (currentPosition === spanElements.length - 1) {
+          if (x === true) {
+            if (spanElements[currentPosition].textContent === event.key) {
+              rightKeys += 1;
+              popup();
+            } else if (spanElements[currentPosition].className === 'red') {
+              if (spanElements[currentPosition].textContent === event.key) {
+                spanElements[currentPosition].className = 'green';
+                popup();
+              }
+            }
+          }
+        }
+        if (spanElements[currentPosition].textContent === event.key) {
+          spanElements[currentPosition].className = 'green';
+          rightKeys += 1;
+        }
+        if (
+          spanElements[currentPosition].textContent !== event.key &&
+          spanElements[currentPosition].innerHTML !== '&nbsp;'
+        ) {
+          spanElements[currentPosition].className = 'red';
+        } else {
+          spanElements[currentPosition].classList.remove('underscore');
+          spanElements[currentPosition + 1].className = 'underscore';
+          currentPosition += 1;
+        }
+        attempts += 1;
+        console.log('Attempts:', attempts);
+        console.log('Right keys', rightKeys);
+      });
+
+      function popup() {
```

Comment:
you don't have to do this now, but i'd put this `popup` function above line 43 where you add the event listener, because your callback function is referencing the `popup` function. It technically works due to javascript hoisting... but if you write it as a function expression it might not work and you'll probably wonder why.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Functions#function_hoisting